lib/console/gateways/gateways.ex

```diff
@@ -105,4 +111,25 @@ defmodule Console.Gateways do
   def change_gateway(%Gateway{} = gateway) do
     Gateway.changeset(gateway, %{})
   end
+
+  def get_gateway_by_unique_identifier(id) do
+    case Base.decode32(id) do
+      {:ok, identifier} ->
+        case Repo.get_by(GatewayIdentifier, unique_identifier: identifier) do
+          gatewayIdentifier = %GatewayIdentifier{} -> gatewayIdentifier |> Repo.preload(:gateway)
+          nil -> :error
+        end
+      :error -> :error
+    end
+  end
+
+  defp create_gateway_identifier do
+    attrs = %{
+      unique_identifier: :crypto.strong_rand_bytes(4)
```

Comment:
do we need to worry about handling the case this random function generates the same identifier?

Reply:
it will raise a database unique constraint error. we should be able to rescue from that and retry n number of times (it'll be a really low probability of happening so a couple retries will be way more than enough)